src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -82,7 +82,8 @@ private function findAndSortTaggedServices($tagName, ContainerBuilder $container
                 } elseif (null === $defaultIndex && $defaultPriorityMethod && $class) {
                     $defaultIndex = PriorityTaggedServiceUtil::getDefault($container, $serviceId, $class, $defaultIndexMethod ?? 'getDefaultName', $tagName, $indexAttribute, $checkTaggedItem);
                 }
-                $index = $index ?? $defaultIndex ?? $defaultIndex = $serviceId;
+                $decorated = $definition->getTag('container.decorator')[0]['id'] ?? null;
+                $index = $index ?? $defaultIndex ?? $defaultIndex = ($decorated ?? $serviceId);
```

Comment:
```suggestion
                $index = $index ?? $defaultIndex ?? $defaultIndex = $decorated ?? $serviceId;
```

Reply:
BTW, I don't understand why there is an assignation here. If I remove it, tests are still okay

---

anyway, fixed, thanks